service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/controller/ReportController.java

```diff
@@ -124,6 +130,14 @@
         }
     }
 
+    private POCDMT000002UK01ClinicalDocument1 getDocumentWithMaxEffectiveDate(List<POCDMT000002UK01ClinicalDocument1> clinicalDocuments) {
+        return Collections.max(clinicalDocuments, Comparator.comparing(doc -> extractEffectiveDateFromClinicalDocument(doc)));
+    }
+
+    private Date extractEffectiveDateFromClinicalDocument(POCDMT000002UK01ClinicalDocument1 clinicalDocument) {
+        return periodMapper.mapPeriod(clinicalDocument.getEffectiveTime()).getStart();
```

Comment:
I think it would be easier to get `clinicalDocument.getEffectiveTime().getValue()` and parse it using `DateUtil.parseToInstantType(effectiveTime)` (check how it's done in MessageHeaderService)